### PR TITLE
FAD-4478 Add mixpanel tracking to tools UI

### PR DESCRIPTION
**Pull request by jonambas**
Events:
- **Viewed Tools Page**
  - Triggered on route change and initial load
  - { route }

- **Tools Login**
  - Triggered after successful login
  - if cookie exists, this always fires on first load

- **Tools Logout**
  - Triggered after logout button click

- **Tools SPF Inspect**
  - Triggered on spf inspect success
  - { domain }

- **Tools Button Click**
  - Triggered on button clicks where track={true}
  - { title, component }
  - SpLoginLink, SaveResultsLink, SpSignUpLink, (not every) ActionLink and Button

- **Tools SPF History Click**
  - Triggered on SPF history list click
  - { domain }

### CODE REVIEW

File: src/actions/mixpanel.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-undef */
```
Comment (jasonrhodes) [inline src/actions/mixpanel.js]: Does it work if you do:
```
import { mixpanel } from window;
```
?

I like being able to track dependencies if possible...

Comment (jasonrhodes) [inline src/actions/mixpanel.js]: err I guess it would be
```
const { mixpanel } = window;
```

File: src/components/button/Button.js [hunk 2]
```diff
@@ -23,41 +25,35 @@ const getLoginSignUpQueryParams = (location) => (
...
-  return (
-    <Link to={to} className={classes}>
-      {children}
-    </Link>
-  );
+  const handleClick = () => {
+    action && action();
+    track && trackButtonClick(children, 'Button');
+  };
+  return <Link to={to} onClick={handleClick} className={classes}>{children}</Link>;
```
Comment (jasonrhodes) [inline src/components/button/Button.js]: This collapse of Link/Action means that all of our buttons in the app will always be `<a>` and never `<button>`, right? Not a huge deal, just clarifying...

File: src/components/button/Button.js [hunk 3]
```diff
@@ -23,41 +25,35 @@ const getLoginSignUpQueryParams = (location) => (
...
 /**
  * Produces an orange link
  * Can be used to link or execute an action
  */
-const ActionLink = (props) => {
-  const { to = null, external = null, onClick = null, title = '', children } = props;
+const DetachedActionLink = (props) => {
+  const { to = null, external = null, onClick = null, title = '', track = false, children, trackButtonClick } = props;
+  const handleClick = () => {
+    !external && onClick && onClick();
```
Comment (jasonrhodes) [inline src/components/button/Button.js]: Do we need to check `!external` here? If someone passes an external URL _and_ an onClick ... that's your own fault ya dummy

Comment (jonambas) [inline src/components/button/Button.js]: I guess not, not sure why it was there. You technically should be able to fire something on click on an external link anyways.

File: src/components/button/Button.js (hunk 2 above)
Comment (jonambas) [inline src/components/button/Button.js]: Yep was trying to reduce redundant code

I think what I can do is:
```
if (!to) {
 return <button onClick={handleClick} className={classes}>{children}</button>;
}
```

**Comment by jonambas** [comment]
So problem - not sure if this is a big deal

After logging in through tools, user is 'identified' to an alias (CID) created in WebUI sign up. Because this alias takes place in WebUI (and uses the webui mp cookie), pre-auth events in tools are not mapped to that CID. All subsequent events would be mapped correctly though.

Me going through login process starting at the bottom:
screen shot 2017-02-28 at 5 46 19 pm

**Pull request merged**